RTech.CoffeeMachine.Api.Tests/BrewCoffee/Controllers/CoffeeMachineControllerTests.cs

```diff
@@ -120,29 +113,110 @@ public async Task GetBrewStatus_WhenCoffeeReady_ShouldReturn200()
         dateTimeProvider.Setup(p => p.GetLocalNow())
             .Returns(nowLocal);
 
-        var response = await GetEndpointResponse(services =>
-        {
-            services.AddSingleton<IDateTimeProvider>(dateTimeProvider.Object);
-        });
+        var response = await GetEndpointResponse(dateTimeProvider: dateTimeProvider.Object);
 
         response.StatusCode.ShouldBe(HttpStatusCode.OK);
         var content = await response.Content.ReadFromJsonAsync<BrewStatusResponse>();
         content.ShouldNotBeNull();
-        content.Message.ShouldBe("Your piping hot coffee is ready");
+        content.Message.ShouldBe(BREW_READY_HOT_MESSAGE);
 
         var offsetStr = DateTime.Now.ToString("zzz").Replace(":", "");
         var expectedPrepared = $"2023-01-02T03:04:05{offsetStr}";
         content.Prepared.ShouldBe(expectedPrepared);
     }
 
-    private Task<HttpResponseMessage> GetEndpointResponse(Action<IServiceCollection> servicesConfiguration)
+    [Fact]
+    public async Task GetBrewStatus_WhenTemperatureRequestThrowsException_ShouldReturn502()
+    {
+        var mockHttp = new MockHttpMessageHandler();
+        mockHttp.When("*")
+            .Throw(new HttpRequestException());
+
+        var httpClient = mockHttp.ToHttpClient();
+        httpClient.BaseAddress = new Uri(MOCKED_HTTPCLIENT_BASE_URL);
+        var response = await GetEndpointResponse(httpClient: httpClient);
+
+        response.StatusCode.ShouldBe(HttpStatusCode.BadGateway);
+        var content = await response.Content.ReadAsStringAsync();
+        content.ShouldBe(string.Empty);
+    }
+
+    [Fact]
+    public async Task GetBrewStatus_WhenTemperatureRequestReturns500_ShouldReturn502()
+    {
+        var mockHttp = new MockHttpMessageHandler();
+        mockHttp.When("*")
+            .Respond(HttpStatusCode.InternalServerError);
+
+        var httpClient = mockHttp.ToHttpClient();
+        httpClient.BaseAddress = new Uri(MOCKED_HTTPCLIENT_BASE_URL);
+        var response = await GetEndpointResponse(httpClient: httpClient);
+
+        response.StatusCode.ShouldBe(HttpStatusCode.BadGateway);
+        var content = await response.Content.ReadAsStringAsync();
+        content.ShouldBe(string.Empty);
+    }
+
+    [Theory]
+    [InlineData(30, BREW_READY_HOT_MESSAGE)]
+    [InlineData(30.01, "Your refreshing iced coffee is ready")]
```

Comment:
`"Your refreshing iced coffee is ready"` is only hard-coded once, so theres little benefit into making it into a const (apart from making the code appear abit more consistent)